### PR TITLE
MISUV-8125: Refactor SessionService.getMongo to take parameter 'JourneyType'

### PR DESCRIPTION
**Pull request by ayubAz**

### CODE REVIEW

File: app/enums/JourneyType/JourneyType.scala [hunk 1]
```diff
@@ -37,3 +39,19 @@ case object Manage extends Operation {
...
+
+case class Opt(optJourney: OptJourney) extends JourneyType {
+  override def toString: String = optJourney.choice
+}
+
+sealed trait OptJourney extends JourneyType {
+  val choice: String
+}
+
+case object In extends OptJourney {
```
Comment (avelx) [inline app/enums/JourneyType/JourneyType.scala:51]: OptInJourney

File: app/enums/JourneyType/JourneyType.scala [hunk 2]
```diff
@@ -37,3 +39,19 @@ case object Manage extends Operation {
...
+  override val choice = "OPTIN"
+}
+
+case object Out extends OptJourney {
```
Comment (avelx) [inline app/enums/JourneyType/JourneyType.scala:55]: OptOutJourney

File: app/enums/JourneyType/JourneyType.scala [hunk 3]
```diff
@@ -37,3 +39,19 @@ case object Manage extends Operation {
 case object Cease extends Operation {
   override val operationType = "CEASE"
 }
+
+case class Opt(optJourney: OptJourney) extends JourneyType {
```
Comment (avelx) [inline app/enums/JourneyType/JourneyType.scala:43]: OptInOutJourney ? not sure, check the context

File: app/enums/JourneyType/JourneyType.scala [hunk 4]
```diff
@@ -18,7 +18,9 @@ package enums.JourneyType
 
 import enums.IncomeSourceJourney.IncomeSourceType
 
-case class JourneyType(operation: Operation, businessType: IncomeSourceType) {
+sealed trait JourneyType
+
+case class IncomeSources(operation: Operation, businessType: IncomeSourceType) extends JourneyType {
```
Comment (avelx) [inline app/enums/JourneyType/JourneyType.scala:23]: this naming convention would create confusion, if you planning to generalise this
you can use something like: IncomeSourceJourneyType or similar to preserver meaning

File: app/enums/JourneyType/JourneyType.scala [hunk 5]
```diff
@@ -18,7 +18,9 @@ package enums.JourneyType
 
 import enums.IncomeSourceJourney.IncomeSourceType
 
-case class JourneyType(operation: Operation, businessType: IncomeSourceType) {
+sealed trait JourneyType
```
Comment (avelx) [inline app/enums/JourneyType/JourneyType.scala:21]: You can name is BaseJourneyType or similar,
as we have to preserver original naming convention which is

File: app/enums/JourneyType/JourneyType.scala (hunk 3 above)
Comment (Antony100) [reply app/enums/JourneyType/JourneyType.scala:43]: The changes you've suggested make it clearer and will implement. `Opt(optJourney)` may be clearer as a result? When this class is called with the proposed changes, we would see `Opt(OptInJourney)` or `Opt(OptOutJourney)`

Comment (avelx) [reply app/enums/JourneyType/JourneyType.scala:43]: I suspect you can leave it as is, its just a bit confusing when I try to read this separately from the actual usage;

**Comment by avelx** [comment]
quite a few imports need to be dropped / optimised, but this can be fixed across the project by running ScalaFix import optimise rules / so no action: https://scalacenter.github.io/scalafix/docs/rules/OrganizeImports.html

**Review by avelx — APPROVED** [review]

**Pull request merged**